pkg/reconciler/ingress/ingress.go

```diff
@@ -107,12 +107,22 @@ func (c *Reconciler) reconcileIngress(ctx context.Context, ing *v1alpha1.Ingress
 	for _, rule := range ing.Spec.Rules {
 		rule := rule
 
-		httproute, probeTargets, err := c.reconcileHTTPRoute(ctx, ingressHash, ing, &rule)
+		workloadRoute, probeTargets, err := c.reconcileWorkloadRoute(ctx, ingressHash, ing, &rule)
 		if err != nil {
 			return err
 		}
 
-		if isHTTPRouteReady(httproute) {
+		// For now, we only generate the redirected HTTPRoute for external visibility,
+		// because currently we do not (yet) support HTTPs for cluster-local domains in net-gateway-api.
```

Comment:
Do we support that for internal encryption and Kourier? :thinking:

Reply:
yes Kourier supports it, net-istio does not need it and net-contour work was started but never finished.
But as this is quite niche and not in the conformance spec, I think we are fine without it here.